### PR TITLE
don't invoke IO manager when MaterializeResult is returned

### PR DESCRIPTION
**Pull request by sryza**

### CODE REVIEW

File: python_modules/dagster/dagster_tests/definitions_tests/test_materialize_result.py [hunk 1]
```diff
@@ -375,7 +375,7 @@ def multi_checks(context: AssetExecutionContext) -> Tuple[MaterializeResult, Mat
     )
 
 
-def test_materialize_result_no_output_typing_calls_io():
+def test_materialize_result_no_output_typing_does_not_call_io():
     """Returning MaterializeResult from a vanilla asset or a multi asset that does not use
     AssetSpecs AND with no return type annotation results in an Any typing type for the
     Output. In this case we emit a warning, then call the I/O manager.
```
Comment (jamiedemaria) [inline python_modules/dagster/dagster_tests/definitions_tests/test_materialize_result.py:381]: Can you update the comment?

**Review by jamiedemaria — COMMENTED** [review]
This looks good to me. One thing I'm unsure about - are there ever times when we want to store ObserveResults? as far as i can tell, this change means all ObserveResults will skip I/O management too (might be good to add that to the title of the PR for easy search later). I don't know enough about the use cases for ObserveResult to say with confidence that we can always skip storage. @smackesey  will probably know

File: python_modules/dagster/dagster_tests/definitions_tests/test_materialize_result.py (hunk 1 above)
Comment (sryza) [reply python_modules/dagster/dagster_tests/definitions_tests/test_materialize_result.py:381]: Oops yes, thanks for catching.

**Comment by sryza** [comment]
> This looks good to me. One thing I'm unsure about - are there ever times when we want to store ObserveResults? 

I can say with confidence that there are not.

**Review by jamiedemaria — APPROVED** [review]
sick

**Pull request merged**